src/registry/registryAccess.ts

```diff
@@ -71,9 +74,31 @@ export class RegistryAccess {
    * @returns An array of metadata type objects that require strict parent folder names
    */
   public getStrictFolderTypes(): MetadataType[] {
-    return Object.values(this.registry.strictDirectoryNames).map(
-      (typeId) => this.registry.types[typeId]
-    );
+    if (!this.strictFolderTypes) {
+      this.strictFolderTypes = Object.values(this.registry.strictDirectoryNames).map(
+        (typeId) => this.registry.types[typeId]
+      );
+    }
+    return this.strictFolderTypes;
+  }
+
+  /**
+   * Query for the types that have the folderContentType property defined.
+   * E.g., reportFolder, dashboardFolder, documentFolder, emailFolder
+   * @see {@link MetadataType.folderContentType}
+   *
+   * @returns An array of metadata type objects that have folder content
+   */
+  public getFolderContentTypes(): MetadataType[] {
+    if (!this.folderContentTypes) {
+      this.folderContentTypes = [];
+      for (const type of Object.values(this.registry.types)) {
```

Comment:
Do you think this is why the registry has so many other properties? To avoid the perf hit of iterating over every type just to find some that have a property defined? Would it make sense to add a `folderContentTypes` property to the registry and just access it directly?

Reply:
By "so many other properties" are you referring to the indexes like suffixes, strictDirectoryNames, and childTypes?  If so, yeah that's for better perf.  There are only 4 folderContentTypes so I don't think it's worth indexing them.  Building the array once makes sense to me though.  I think RegistryAccess could use more memoization logic like this.

Reply:
yeah exactly. I'd prefer not to iterate over the whole registry to find four entries, multiple times... but we could do this optimization later

Reply:
It's only done once, not multiple times.